expression/builtin_compare_vec.go

```diff
@@ -715,11 +715,53 @@ func (b *builtinNEStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column
 }
 
 func (b *builtinGreatestTimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinGreatestTimeSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	dst, err := b.bufAllocator.get(types.ETTimestamp, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(dst)
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+	dst.ResizeTime(n, false)
+	dstTimes := dst.Times()
+	var argTime types.Time
+	for j := 0; j < len(b.args); j++ {
+		if err := b.args[j].VecEvalString(b.ctx, input, result); err != nil {
+			return err
+		}
+		for i := 0; i < n; i++ {
+			if result.IsNull(i) || dst.IsNull(i) {
+				dst.SetNull(i, true)
+				continue
+			}
+			argTime, err = types.ParseDatetime(sc, result.GetString(i))
+			if err != nil {
+				if err = handleInvalidTimeError(b.ctx, err); err != nil {
+					return err
+				}
+				dst.SetNull(i, true)
+				continue
+			}
+			if argTime.Compare(dstTimes[i]) > 0 {
+				dstTimes[i] = argTime
+			}
+		}
+		result.ReserveString(n)
+	}
+
+	for i := 0; i < n; i++ {
+		if dst.IsNull(i) {
+			result.AppendNull()
+		} else {
+			result.AppendString(dstTimes[i].String())
+		}
+	}
```

Comment:
Please use `Column.CopyConstruct`.

Reply:
The `dst` is timestamp-type, but the `result` is string-type. According to the source code of  `Column.CopyConstruct`, maybe it suits only for the same type colume.